internal/controllers/utils/utils.go

```diff
@@ -175,17 +260,33 @@ func BuildServerContainerArgs(orano2ims *oranv1alpha1.ORANO2IMS,
 		return
 	}
 
+	// ResourceServer:
 	if serverName == ORANO2IMSResourceServerName {
+		searchAPI := orano2ims.Spec.ResourceServerConfig.BackendURL
+		if searchAPI == "" {
+			searchAPI, err = getSearchAPI(ctx, c, orano2ims)
+			if err != nil {
+				return nil, err
+			}
+		}
+
 		result = slices.Clone(ResourceServerArgs)
+
+		// Add the cloud-id and backend-url args:
 		result = append(
 			result,
 			fmt.Sprintf("--cloud-id=%s", orano2ims.Spec.CloudId),
-			fmt.Sprintf("--backend-url=%s", orano2ims.Spec.SearchAPIBackendURL),
-			fmt.Sprintf("--backend-token=%s", orano2ims.Spec.BackendToken))
+			fmt.Sprintf("--backend-url=%s", searchAPI))
 
-		return
+		// Add the token arg:
+		result = append(
+			result,
+			GetBackendTokenArg(orano2ims.Spec.ResourceServerConfig.BackendToken))
```

Comment:
Requests to the resource server didn't work for me when the default token file was used ("/run/secrets/kubernetes.io/serviceaccount/token").
This doesn't have to block the merge as the resource server token can be included in the spec.

Reply:
What do you mean when you shay that it doesn't work? What error does it write to the log? If you get an error then we may have a problem in how we pass that token to the server, because in theory any service account has permission to use the ACM search API. You can find more details about how that is checked here:

https://github.com/stolostron/search-v2-api/blob/main/pkg/rbac/authnMiddleware.go

If there is no error, but you don't get the expected results, then it will probably be because the service account that you are using doesn't have the right permissions. The ACM search API uses the same permissions than the rest of the cluster. You can find more details about how that is implemented here:

https://github.com/stolostron/search-v2-api/blob/main/pkg/rbac/authzMiddleware.go

If I understand correctly we create a service account for each service. For the resource server is named `resource-server`, I think. That service account won't have any permissions. For example, when you send a request like this:

```
GET /o2ims-infrastructureInventory/v1/resourcePools
```

Our resource server will try to use the search API to retrieve the set of `ManagedClusters` (this is the mapping we currently do: a resource pool per `ManagedCluster`, @danielerez correct me if I am wrong) and will receive nothing because the `resource-server` account doesn't have permissions to get `ManagedCluster` objects.

In the documentation [here](https://github.com/openshift-kni/oran-o2ims/blob/main/docs/dev/env_acm.md#create-a-token-for-accessing-the-api) @danielerez suggests to use the `oauth-apiserver-sa` service account from the `openshift-oauth-apiserver` namespace. That works because that service account is the one used by the ACM search API itself, and it has all the permissions. Instead of that I think that our operator should add to our `resource-server` service account all the required permissions, creating a role and a role binding for that. I think that role will need read access to `ManagedClusters` and `Hosts`. @danielerez please clarify with @irinamihai what are the required permissions.

If my analysis is right, that work ^ looks complicated enough to deserve a separate pull request. Please merge this first, and then open another one for that.

Reply:
Thank you for all the info, @jhernand. I apologize, I should have provided more details.
It doesn't work in the sense that an empty list is always returned by the server. I did multiple runs and I noticed this is the behavior when the token is not correct.
I will look more into it and discuss with @danielerez for the future PR you've suggested.

I will merge this since we can specify the backendToken for the resourceServer separately.

Reply:
So about the access permissions, in the resource server we query two kinds of objects: [Cluster](https://github.com/openshift-kni/oran-o2ims/blob/b1a7a90ef4f0de09955a90d7f8aceebb0bbe6e68/internal/cmd/server/start_resource_server.go#L523) and [Node](https://github.com/openshift-kni/oran-o2ims/blob/b1a7a90ef4f0de09955a90d7f8aceebb0bbe6e68/internal/cmd/server/start_resource_server.go#L535).
Then I guess it could indeed be enough to have read access for these, but will have to check if anything else is maybe required by the search API.
Regarding the empty list using an invalid token, it does look like a bug. I would expect a StatusUnauthorized (401) error in such scenario. Will have to look into it.